src/main/java/userReg/UserProfile.java

```diff
@@ -0,0 +1,80 @@
+package userReg;
+
+import data_access_storage.RequestModel;
+import data_access_storage.UserRepo;
+import data_access_storage.UserRepoManager;
+import entities.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public class UserProfile extends User implements  ProfileManager{
+
+    String email;
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    String password;
+
+    public  UserProfile(String email,String password,String name, Gender gender, int age,
+                        float height, String programOfStudy,
+                        Hobbies hobby, String selfIntro, Gender interestedIn){
+        super(name,gender,age,height,programOfStudy,hobby,selfIntro,interestedIn);
+        this.email=email;
+        this.password=password;
+
+    }
+
+
+    @Override
+    public boolean checkEmail() {
+        String email=this.email;
+        UofTStudentEmailVerifier usev=new UofTStudentEmailVerifier();
+        return usev.verify(email);
+    }
+
+    @Override
+    public boolean existEmail() {
+        UserRepoManager users;
+        try{
+            File csvFile = new File("src/main/resources/testFile.csv");
+            users = new UserRepo(csvFile);
+        }catch (IOException e) {
+            throw new RuntimeException("Could not create file.");
+        }
+       return  users.existsByEmail(this.email);
+
+    }
+
+    @Override
+    public void save(){
+        UserRepoManager users;
+        try{
+            File csvFile = new File("src/main/resources/testFile.csv");
+            users = new UserRepo(csvFile);
+        }catch (IOException e) {
+            throw new RuntimeException("Could not create file.");
+        }
+
+        UserAccount ua =new UserAccount(email,password,this);
+        RequestModel rm=new RequestModel(ua);
+        users.save(rm);
+
+
+    }
+}

```

Comment:
1. This class violates dependence rules required by clean architecture. For example, your UserProfile extends User. Since User is in the entities layer, your UserProfile is in the entities layer as well. That means UserProfile should know nothing about things in Layer Frameworks & Drivers. However, your UserProfile calls our database directly. This is a violation of clean architecture. Many revisions need to be made.  Please have a look at package email_request for code structure. 
3.  Class extension is for "is a " relationship. For example, a dog is an animal.  The relationship between UserProfile and User is not a "a" relationship.   The reason why UserProfile extends User here is probably because you want to use some instance variables and methods from User class. However, our class in Entities UserAccount should have what you all need.  If not, please let me know. Additional methods or instance variables can be added to UserAccount.